server/gitrest/src/routes/git/refs.ts

```diff
@@ -28,24 +31,57 @@ async function getRefs(repoManager: utils.RepositoryManager, owner: string, repo
     return refsP.map((ref) => refToIRef(ref));
 }
 
-async function getRef(repoManager: utils.RepositoryManager, owner: string, repo: string, refId: string): Promise<IRef> {
+async function getRef(
+    repoManager: utils.RepositoryManager,
+    owner: string,
+    repo: string,
+    refId: string,
+    externalStorageManager: IExternalStorageManager): Promise<IRef> {
     const repository = await repoManager.open(owner, repo);
-    const ref = await git.Reference.lookup(repository, refId, undefined);
-    return refToIRef(ref);
+    try {
+        const ref = await git.Reference.lookup(repository, refId, undefined);
+        return refToIRef(ref);
+    } catch (err) {
+        winston.error(`getRef error: ${err}`);
+        // Lookup external storage if commit does not exist.
+        const fileName = refId.substring(refId.lastIndexOf("/") + 1);
+        // If file does not exist or error trying to look up commit, return the original error.
+        try {
+            const result = await externalStorageManager.read(repo, fileName);
+            if (result === false) {
+                return err;
```

Comment:
Should we reject the promise here?

Reply:
Looks like lots of the comments are marked as resolved but the change has not been made?